### ISSUE TITLE
FHIR Search DISTINCT and GROUP BY

### ISSUE THREAD

**Issue by alkutsy** [hapifhir/hapi-fhir#3026]
Hello there,

**Description**
Is it possible to do GROUP BY or DISTINCT queries? For instance I need to generate statistic: number of Persons per disease. I use Condition resource  that contains disease code and Person ID. 

**Expected behavior**
DISTINCT query return all unique disease codes from Condition resource. 
GROUP BY able to group persons by disease code.

Condition example:

`"resource": {
      "resourceType": "Condition",
      "id": "67379361",
      "meta": {
        "extension": [ {
          "url": "http://hapifhir.io/fhir/StructureDefinition/resource-meta-source",
          "valueUri": "#kwjKm4uIyxSnKQz3"
        } ],
        "versionId": "1",
        "lastUpdated": "2021-09-15T08:33:46.857+00:00",
        "profile": [ "http://standardhealthrecord.org/fhir/StructureDefinition/shr-problem-Problem" ]
      },
      "clinicalStatus": "active",
      "verificationStatus": "confirmed",
      "code": {
        "coding": [ {
          "system": "http://snomed.info/sct",
          "code": "10509002",
          "display": "Acute bronchitis (disorder)"
        } ],
        "text": "Acute bronchitis (disorder)"
      },
      "subject": {
        "reference": "urn:uuid:dbad7fae-b821-4f86-9107-5eaf61310de6"
      },
      "context": {
        "reference": "urn:uuid:2e7299b6-d950-4293-aaac-ca131c7abe9e"
      },
      "onsetDateTime": "2011-06-04T13:45:03-04:00",
      "abatementDateTime": "2011-06-21T09:25:04-04:00",
      "assertedDate": "2011-06-04"
    }`

**Environment (please complete the following information):**
 - HAPI FHIR Version: 5.5.1
 - OS: Ubuntu 21.04
 - Browser: google chrome

Thanks in advance

**Comment by dajaffe** [comment]
Came across this looking for a similar use case; I'm looking to query all Condition resources with a GROUP BY + COUNT to identify patients who have the same Condition (grouped by unique coding) recorded multiple times. Is there a way to format a FHIR query for this, or is executing raw queries against the backing database necessary?

**Comment by jamesagnew** [comment]
I don't believe there is anything in the FHIR REST API that does anything equivalent to the SQL group by clause.

[HFQL](https://hapi.fhir.org/hfql) might be an answer though?
